graylog2-server/src/main/java/org/graylog2/lookup/adapters/CSVFileDataAdapter.java

```diff
@@ -99,13 +105,20 @@ public void doStart() throws Exception {
         if (!pathChecker.fileIsInAllowedPath(Paths.get(config.path()))) {
             throw new IllegalStateException(ALLOWED_PATH_ERROR);
         }
+        if (config.canBeOverridden() && !pathChecker.fileIsInAllowedPath(Paths.get(Objects.requireNonNull(config.overridePath())))) {
```

Comment:
Readability-wise, _pathChecker.filesInAllowedPath(Paths.get(Objects.requireNonNull(config.overridePath())_ is too complex.  Also, **Paths.get(Objects.requireNonNull(config.overridePath()** can be evaluated twice, depending on the state of **config.canBeOverridden()**.    It may be best if **Paths.get(Objects.requireNonNull(config.overridePath()** is evaluated and saved to a variable once.

Also, the same operation is found in doRefresh(..)